src/utils/workflow-utils.ts

```diff
@@ -62,3 +70,133 @@ export const modifyDiscoveredDraftSpec = async (
         supabaseConfig?.lastPubId
     );
 };
+
+interface CollectionData {
+    [key: string]: {
+        spec: any;
+        expect_pub_id: string | null;
+    };
+}
+
+export const getBoundCollectionData = (
+    boundCollections: string[],
+    draftSpecData: DraftSpecQuery[],
+    liveSpecData: LiveSpecsExtQuery_ByCatalogNames[]
+): CollectionData => {
+    let collectionData: CollectionData = {};
+
+    boundCollections.forEach((collection) => {
+        const draftSpecQuery = draftSpecData.find(
+            ({ catalog_name }) => catalog_name === collection
+        );
+        const liveSpecQuery = liveSpecData.find(
+            ({ catalog_name }) => catalog_name === collection
+        );
+
+        if (draftSpecQuery) {
+            const { spec, expect_pub_id } = draftSpecQuery;
+
+            collectionData = {
+                ...collectionData,
+                [collection]: { spec, expect_pub_id },
+            };
+        } else if (liveSpecQuery) {
+            const { spec, last_pub_id } = liveSpecQuery;
+
+            collectionData = {
+                ...collectionData,
+                [collection]: { spec, expect_pub_id: last_pub_id },
+            };
+        } else {
+            console.log(`Could not find any collection data for ${collection}`);
```

Comment:
Presenting this as a top-level warning that is surfaced post-discovery felt like an _odd_ UX. If that is the desired path forward, I can throw the code back in and perhaps a message that provides proper context would ease the awkwardness I sensed.

Reply:
What steps does the user have to take (if any) to fix the issue?

Reply:
The only option would be to reapply their changes for the identified collections and rediscover.

Reply:
That seems fine... let's wait until we have next button in there so it is easier for them to sort out what needs to happen.

We should add a LogRocket event here though.